sdktests/server_side_persistence_consul.go

```diff
@@ -0,0 +1,121 @@
+package sdktests
+
+import (
+	"fmt"
+	"strings"
+
+	consul "github.com/hashicorp/consul/api"
+	"github.com/launchdarkly/sdk-test-harness/v2/servicedef"
+)
+
+type ConsulPersistentStore struct {
+	consul *consul.Client
+}
+
+// {{{ PersistentStore implementation
+
+func (c ConsulPersistentStore) DSN() string {
+	// TODO: Fix this address lookup
+	return fmt.Sprintf("%s", consul.DefaultConfig().Address)
+}
+
+func (c *ConsulPersistentStore) Type() servicedef.SDKConfigPersistentType {
+	return servicedef.Consul
+}
+
+func (c *ConsulPersistentStore) Reset() error {
+	_, err := c.consul.KV().DeleteTree("/", nil)
+	return err
+}
+
+func (c *ConsulPersistentStore) Get(prefix, key string) (string, bool, error) {
+	kv := c.consul.KV()
+
+	pair, _, err := kv.Get(prefix+"/"+key, nil)
+	if err != nil || pair == nil {
+		return "", false, err
+	}
+
+	return string(pair.Value), true, nil
+}
+
+func (c *ConsulPersistentStore) GetMap(prefix, key string) (map[string]string, error) {
+	kv := c.consul.KV()
+	pairs, _, err := kv.List(prefix+"/"+key, nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("list failed for %s: %s", key, err)
+	}
+
+	results := make(map[string]string)
+	for _, pair := range pairs {
+		flagKey := strings.TrimPrefix(pair.Key, prefix+"/"+key+"/")
+		results[flagKey] = string(pair.Value)
+	}
+	return results, nil
+}
+
+func (c *ConsulPersistentStore) WriteMap(prefix, key string, data map[string]string) error {
+	kv := c.consul.KV()
+
+	// Start by reading the existing keys; we will later delete any of these that weren't in allData.
+	pairs, _, err := kv.List(prefix, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get existing items prior to Init: %s", err)
+	}
+	oldKeys := make(map[string]bool)
```

Comment:
If `oldKeys` is meant to represent a set, I find it clearer to use `map[string]struct{}`. That way, there's no need to check `v` (on line 81 of this diff).

On line 77, you'd `delete` from the set.

Reply:
Yes, I thought about this when I was copying this work over from the actual implementations. I went back and forth on whether consistency was better.

I'll just fix it here.